tests/bgp/test_bgp_sentinel.py

```diff
@@ -0,0 +1,358 @@
+            ipv4_res = ptfhost.shell("ping {} -c 3 -I backplane".format(lo_ipv4_addr), module_ignore_errors=True)
+            if ipv4_res['rc'] != 0:
+                ptfhost.shell("ip route del {} via {}".format(lo_ipv4_addr, v['ipv4']), module_ignore_errors=True)
+            else:
+                ipv4_nh = v['ipv4']
+
+        if ipv6_nh is None:
+            ptfhost.shell("ip route add {} via {}".format(lo_ipv6_addr, v['ipv6']), module_ignore_errors=True)
+            ipv6_res = ptfhost.shell("ping {} -c 3 -I backplane".format(lo_ipv6_addr), module_ignore_errors=True)
+            if ipv6_res['rc'] != 0:
+                ptfhost.shell("ip route del {} via {}".format(lo_ipv6_addr, v['ipv6']), module_ignore_errors=True)
+            else:
+                ipv6_nh = v['ipv6']
+
+    return ipv4_nh, ipv6_nh
+
+
+@pytest.fixture(scope="module")
+def dut_lo_addr(rand_selected_front_end_dut):
+    duthost = rand_selected_front_end_dut
+    lo_facts = duthost.setup()['ansible_facts']['ansible_Loopback0']
+    lo_ipv4_addr, lo_ipv6_addr = lo_facts['ipv4']['address'], None
+    for item in lo_facts['ipv6']:
+        if item['address'].startswith('fe80'):
+            continue
+        lo_ipv6_addr = item['address']
+        break
+    return lo_ipv4_addr, lo_ipv6_addr
+
+
+@pytest.fixture(scope="module")
+def dut_setup_teardown(rand_selected_front_end_dut, tbinfo, dut_lo_addr):
+    duthost = rand_selected_front_end_dut
+    lo_ipv4_addr, lo_ipv6_addr = dut_lo_addr
+    ipv4_subnet, ipv6_subnet, spine_bp_addr = get_dut_listen_range(tbinfo)
+    ptf_bp_v4 = tbinfo['topo']['properties']['configuration_properties']['common']['nhipv4']
+    ptf_bp_v6 = tbinfo['topo']['properties']['configuration_properties']['common']['nhipv6'].lower()
+
+    # render template and write to DB, check running configuration for BGP_sentinel
+    bgp_sentinelv4_tmpl = Template(BGP_SENTINEL_TMPL)
+    duthost.copy(content=bgp_sentinelv4_tmpl.render(v4_listen_range=json.dumps([ipv4_subnet, ptf_bp_v4 + '/32']),
+                                                    v4_src_address=lo_ipv4_addr,
+                                                    v6_listen_range=json.dumps([ipv6_subnet, ptf_bp_v6 + '/128']),
+                                                    v6_src_address=lo_ipv6_addr),
+                 dest=BGPSENTINEL_CONFIG_FILE)
+    duthost.shell("sonic-cfggen -j {} -w".format(BGPSENTINEL_CONFIG_FILE))
+
+    # Once https://github.com/sonic-net/sonic-buildimage/pull/14844 is merged into sonic 202205, we can remove this
+    duthost.shell("vtysh -c \"configure terminal\" -c \"ipv6 nht resolve-via-default\"")
+
+    yield lo_ipv4_addr, lo_ipv6_addr, spine_bp_addr, ptf_bp_v4, ptf_bp_v6
+
+    # Cleanup bgp monitor
+    duthost.run_sonic_db_cli_cmd("CONFIG_DB del 'BGP_SENTINELS|BGPSentinel'", asic_index='all')
+    duthost.run_sonic_db_cli_cmd("CONFIG_DB del 'BGP_SENTINELS|BGPSentinelV6'", asic_index='all')
```

Comment:
Will "CONFIG_DB del 'BGP_SENTINELS|BGPSentinel'" cause bgp container restart ? or we need specify it in script ?

Reply:
It will not cause bgp container restart.